Source/SaveExtension/Private/Serialization/SlotDataTask_Loader.cpp

```diff
@@ -414,8 +415,20 @@ void USlotDataTask_Loader::PrepareLevel(const ULevel* Level, FLevelRecord& Level
 		ActorsToSpawn.Shrink();
 	}
 
-	// Create Actors that doesn't exist now but were saved
-	RespawnActors(ActorsToSpawn, Level);
+	UWorld* const World = GetWorld();
+	CollectGarbage(RF_NoFlags);
+	FTimerHandle DummyHandle;
+	// Collect garbage to avoid spawning issues
+	World->GetTimerManager().SetTimer(
+		DummyHandle,
+		[this, ActorsToSpawn, Level]
+		{
+			// Create Actors that doesn't exist now but were saved
+			this->RespawnActors(ActorsToSpawn, Level);
+		},
+		0.01f,
+		false
+	);
```

Comment:
Not proud of this change, but it get's the job done. 

Basically if we try loading the scene right after destroying an actor that's going to get respawned in the save, it won't get respawned. The first thing we need to do is `CollectGarbage` to ensure the actor is fully deleted. If wee try spawning the actors immediately afterward, however, the actors still won't spawn. Spawned actors with the same name as `PendingKill` actors, for whatever reason, are _also_ thought of as `PendingKill`, causing spawning to fail. This small delay simply ensures garbage has been collected and the actors are no longer marked as 'PendingKill' so that the new actors can spawn in.